Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts

```diff
@@ -22,25 +25,43 @@ export class ContainerBasedDeploymentUtility {
 
     public async deployWebAppImage(properties: any): Promise<void> {
         let imageName: string = properties["ImageName"];
-        tl.debug("Deploying an image " + imageName + " to the webapp " + this._appService.getName());
-
+        let configFilePath: string = properties["ConfigFilePath"];
+        let isMultiContainer: boolean = properties["isMultiContainer"];
         let isLinuxApp: boolean = properties["isLinuxContainerApp"];
+        let updatedConfigFilePath: string;
+        tl.debug("Deploying image " + imageName + " to the webapp " + this._appService.getName());
+
+        if(isMultiContainer) {
+            updatedConfigFilePath = this.updateImagesInConfigFile(configFilePath, imageName);
+
+            // uploading log file
+            console.log(`##vso[task.uploadfile]${updatedConfigFilePath}`);
+        }
+
         tl.debug("Updating the webapp configuration.");
-        await this._updateConfigurationDetails(properties["ConfigurationSettings"], properties["StartupCommand"], imageName, isLinuxApp);
+        await this._updateConfigurationDetails(properties["ConfigurationSettings"], properties["StartupCommand"], imageName, isLinuxApp, isMultiContainer, configFilePath);
```

Comment:
Should you be using updatedFilePath here? I couldnt find where you updated original file content. 
It seems this will deploy provided config file content without transformation.

Reply:
It is done.